src/Microsoft.AspNet.Hosting/Internal/HostingApplication.cs

```diff
@@ -46,28 +46,28 @@ public Context CreateContext(IFeatureCollection contextFeatures)
             {
                 HttpContext = httpContext,
                 Scope = scope,
-                StartTick = startTick,
+                StartTimestamp = startTick,
             };
         }
 
         public void DisposeContext(Context context, Exception exception)
         {
             var httpContext = context.HttpContext;
-            var currentTick = Environment.TickCount;
-            _logger.RequestFinished(httpContext, context.StartTick, currentTick);
+            var currentTimestamp = context.StartTimestamp != 0 ? Stopwatch.GetTimestamp() : 0;
+            _logger.RequestFinished(httpContext, context.StartTimestamp, currentTimestamp);
 
             if (exception == null)
             {
                 if (_diagnosticSource.IsEnabled("Microsoft.AspNet.Hosting.EndRequest"))
                 {
-                    _diagnosticSource.Write("Microsoft.AspNet.Hosting.EndRequest", new { httpContext = httpContext, tickCount = currentTick });
+                    _diagnosticSource.Write("Microsoft.AspNet.Hosting.EndRequest", new { httpContext = httpContext, tickCount = currentTimestamp });
```

Comment:
Renamed to timestamp; also enabled the checking if either the diagnostics is enabled or the logging (rather than just logging)